pkg/controller/controller.go

```diff
@@ -963,11 +963,6 @@ func (p *csiProvisioner) getPVCSource(ctx context.Context, claim *v1.PersistentV
 		return nil, fmt.Errorf("the requested PVC (%s) storageclass cannot be empty", claim.Name)
 	}
 
-	if *sourcePVC.Spec.StorageClassName != *claim.Spec.StorageClassName {
```

Comment:
Should we add some validation that known first class attributes like fstype are compatible?

Reply:
I thought the consensus was that things like that were driver specific and should be handled by the, eg, pd or filestore driver (where the test might be different).

Even though snapshots can't validate anything, the only problems we've seen are on the other side where the over-validation of enforcing storageclasses to be identical prevent cloning. So I'm inclined to have more relaxed validation which ultimately unblocks workloads.

Reply:
fstype is a first class field in Kubernetes/CSI, and it's information that cannot be easily obtained in the current CSI CreateVolume call since it's not an attribute that block storage really cares about.

Reply:
So to summarize, we should validate Fstype in the provisioner because right now we don't have way to validate that the fstype of the source block volume is compatible with the fstype of the destination block volume. `createVolumeRequest.GetVolumeCapability().GetBlock()` doesn't have any field `FsType`.  I believe we have a way to validate fstype for Filesystem volumes, `createVolumeRequest.GetVolumeCapability().GetMount().FsType`, correct?. In the Kubernetes Sig storage meeting, someone threw out the idea of adding documentation to say that some params like fstype may not be compatible if you have a different storage class. Could we do this instead of validating?

Reply:
>  I believe we have a way to validate fstype for Filesystem volumes, `createVolumeRequest.GetVolumeCapability().GetMount().FsType`, correct

Yes, a CSI driver can get the destination fsType as you suggest above. But it has no way to get the source volume fsType, it gets only `volume_id` of the source volume / `shapshot_id` of the source snapshot.

I don't have a strong opinion about this PR. If the external-provisioner can't check fsType when restoring snapshots (they don't have fsType), why would it check it in cloning? Should we add fsType to snapshots?

Reply:
Sounds good to me. My only ask is we make sure cloning and snapshot documentation has a note that the user needs to make sure source and target parameters are compatible.

Reply:
+1